src/file.cpp

```diff
@@ -937,6 +937,20 @@ namespace libtorrent
 		branch.append(leaf.data(), leaf.size());
 	}
 
+	std::string combine_path(std::string const& ls, std::string const& rs)
+	{
+		string_view lhs{ls};
+		string_view rhs{rs};
+		return combine_path(lhs, rhs);
+	}
+
+	std::string combine_path(char const* ls, char const* rs)
```

Comment:
yes it's necessary, comine_path("a", "b") shows ambiguity (can not decide which of string_view or std::string cont&  to use)